sounddevice.py

```diff
@@ -2289,13 +2289,40 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Attributes
+    ----------
+    args
+        A variable length tuple containing the following elements when
+        available:
+
+        1) The user error message
+        2) The PortAudio ``PaErrorCode`` value
+        3) A 3-tuple containing the host API index, host error code, and the
+           host error message or a zero length string if no message is
+           available
```

Comment:
The thing about the zero length string seems a bit verbose.
Is it really relevant to mention it?
Does this even happen?

If yes, wouldn't something like this be easier to read?

    ... and the host error message (which may be an empty string)

A combination of "and" and "or" is often hard to understand in natural language.
In fact "and" binds stronger than "or", which makes the sentence actually wrong!